crates/biome_js_analyze/src/lint/nursery/use_semantic_elements.rs

```diff
@@ -59,25 +63,83 @@ impl Rule for UseSemanticElements {
             let is_not_interative =
                 AriaRoles.is_not_interactive_element(element_name, extract_attributes);
             if is_not_interative {
-                return Some(AnyJsxAttribute::from(attr));
+                let static_value = attr.as_static_value().unwrap();
+                let role_value = static_value.as_string_constant().unwrap();
+                let candidates = AriaRoles.get_corresponding_element(role_value);
+
+                if let Some(elements) = candidates {
+                    let mut result_elements: Vec<String> = vec![];
+                    let mut result_attributes: Vec<(String, String)> = vec![];
+
+                    for element in elements {
+                        // Get only first attribute
+                        let attributes = element.1.first();
+
+                        result_elements.push(element.0.to_string());
+
+                        if let Some(attribute) = attributes {
+                            result_attributes
+                                .push((attribute.0.to_string(), attribute.1.to_string()));
+                        } else {
+                            result_attributes.push((String::new(), String::new()));
+                        }
+                    }
+
+                    return Some((
+                        AnyJsxAttribute::from(attr),
+                        Some(result_elements),
+                        Some(result_attributes),
+                    ));
+                } else {
+                    return Some((AnyJsxAttribute::from(attr), None, None));
+                }
             }
         }
 
         None
     }
 
     fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let alternative_elements = state.clone().1;
+        let alternative_attributes = state.clone().2;
```

Comment:
I don't think you need a clone:
```suggestion
        let (alternative_elements, alternative_attributes) = state;
```